pool_test.go

```diff
@@ -199,3 +199,100 @@ func TestPoolClose(t *testing.T) {
 		t.Errorf("Returned connection was not closed, underlying connection is not in shutdown state")
 	}
 }
+
+func TestContextCancelation(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
We're running in tests so it is expected we're initialising contexts in this scope. 

```suggestion
	ctx, cancel := context.WithCancel(context.Background())
```